app/test_main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "valid_url_value, connection_value, expected_value",
+    [
+        (False, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (True, False, "Not accessible"),
+        (True, True, "Accessible")
+    ]
+)
+def test_access_google_page(
+        valid_url_value: bool,
+        connection_value: bool,
+        expected_value: str
+) -> None:
+    with (
+        mock.patch(
+            "app.main.valid_google_url",
+            return_value=valid_url_value
+        ),
+        mock.patch(
+            "app.main.has_internet_connection",
+            return_value=connection_value)):
```

Comment:
Apply mocks as decorators and pass them to the function instead to avoid creating a context manager got the whole body of the function

Reply:
This comment is not implemented in two change requests in line. In case you face errors, please visit Q&A to discuss it with mentor. In other case, please apply two mocks as two function decorators to the `test_access_google_page`.